requirejs/text fails with webgme 1.7.x

The webgme-cli command line tool fails with WebGME 1.7.x :

`$ webgme ls`

`...
Error: Tried loading "text" at /usr/lib/node_modules/webgme-cli/lib/text.js then tried node's require("text") and it failed with error: Error: Cannot find module 'text'
...
`

The problem can be temporary fixed by downgrading the webgme dependency of the cli tool to 1.6.x.

--- a/package.json
+++ b/package.json
@@ -43,7 +43,7 @@
     "ramda": "^0.15.1",
     "requirejs": "^2.1.18",
     "rimraf": "^2.4.0",
-    "webgme": "^1.1.0"
+    "webgme": "^1.7.0"
   },
   "devDependencies": {
     "chai": "^3.0.0",

--- a/src/AddonManager.js
+++ b/src/AddonManager.js
@@ -19,7 +19,8 @@ var _ = require('lodash'),
     AddonGenerator = require('./shim/AddonGenerator'),
     Enableable = require('./mixins/Enableable/Enableable'), 
     PluginHelpers = require('./shim/PluginHelpers'),
-    RAW_CONFIG = AddonGenerator.prototype.getConfigStructure();
+    metadata = require('webgme/src/plugin/coreplugins/AddOnGenerator/metadata.json'),
+    RAW_CONFIG = metadata.configStructure;
 
 var AddonManager = function(logger) {
     ComponentManager.call(this, 'addon', logger);

--- a/src/DecoratorManager.js
+++ b/src/DecoratorManager.js
@@ -18,8 +18,7 @@ var _ = require('lodash'),
     ComponentManager = require('./ComponentManager'),
     DecoratorGenerator = require('./shim/DecoratorGenerator'),
     Enableable = require('./mixins/Enableable/Enableable'), 
-    PluginHelpers = require('./shim/PluginHelpers'),
-    RAW_CONFIG = DecoratorGenerator.prototype.getConfigStructure();
+    PluginHelpers = require('./shim/PluginHelpers');
 
 var DecoratorManager = function(logger) {
     ComponentManager.call(this, 'decorator', logger);

--- a/src/LayoutManager.js
+++ b/src/LayoutManager.js
@@ -17,8 +17,7 @@ var _ = require('lodash'),
     utils = require('./utils'),
     ComponentManager = require('./ComponentManager'),
     LayoutGenerator = require('./shim/LayoutGenerator'),
-    PluginHelpers = require('./shim/PluginHelpers'),
-    RAW_CONFIG = LayoutGenerator.prototype.getConfigStructure();
+    PluginHelpers = require('./shim/PluginHelpers');
 
 var LayoutManager = function(logger) {
     ComponentManager.call(this, 'layout', logger);

--- a/src/PluginManager.js
+++ b/src/PluginManager.js
@@ -19,7 +19,8 @@ var _ = require('lodash'),
     PluginGenerator = require('./shim/PluginGenerator'),
     Enableable = require('./mixins/Enableable/Enableable'), 
     PluginHelpers = require('./shim/PluginHelpers'),
-    RAW_CONFIG = PluginGenerator.prototype.getConfigStructure();
+    metadata = require('webgme/src/plugin/coreplugins/PluginGenerator/metadata.json'),
+    RAW_CONFIG = metadata.configStructure;
 
 var PluginManager = function(logger) {
     ComponentManager.call(this, 'plugin', logger);

--- a/src/VisualizerManager.js
+++ b/src/VisualizerManager.js
@@ -18,8 +18,7 @@ var _ = require('lodash'),
     ComponentManager = require('./ComponentManager'),
     VisualizerGenerator = require('./shim/VisualizerGenerator'),
     Enableable = require('./mixins/Enableable/Enableable'), 
-    PluginHelpers = require('./shim/PluginHelpers'),
-    RAW_CONFIG = VisualizerGenerator.prototype.getConfigStructure();
+    PluginHelpers = require('./shim/PluginHelpers');
 
 var VisualizerManager = function(logger) {
     ComponentManager.call(this, 'visualizer', logger);

--- a/src/shim/PluginGenerator.js
+++ b/src/shim/PluginGenerator.js
@@ -29,7 +29,7 @@ PluginGenerator.prototype.main = function(callback) {
     var self = this;
     WebGMEPluginGenerator.prototype.main.call(this, function(e, result) {
         if (e) {
-            this.logger.error(e);
+            self.logger.error(e);
             callback(e);
         }
 

--- a/src/utils.js
+++ b/src/utils.js
@@ -338,6 +338,7 @@ var loadPaths = function(requirejs) {
         nodeRequire: require,
         baseUrl: __dirname,
         paths: {
+            text: __dirname+'/../node_modules/webgme/src/client/lib/require/require-text/text',
             coreplugins: __dirname+'/../node_modules/webgme/src/plugin/coreplugins',
             plugin: __dirname + '/../node_modules/webgme/src/plugin',
             common: __dirname + '/../node_modules/webgme/src/common',

